### ISSUE TITLE
BDA-06 should expect 401 along with 400 as a valid http response code (FI-1313)

### ISSUE THREAD

**Issue by sagarshah1983** [onc-healthit/inferno-program#345]
**Subject of the issue**
For multiple patient API tests, when I try positive scenarios using inferno, I found that BDA-06 was still failing with this errors.
Message: Bad response code: expected 400, but found 401
Response Body from Open ID Provider:
```
{
  "error": "invalid_client",
  "error_description": "Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method). Claim 'iss' from 'client_assertion' must match the 'client_id' of the OAuth 2.0 Client."
}
```

Response Headers:
```
date: Tue, 17 Aug 2021 18:56:26 GMT
content_type: application/json;charset=ISO-8859-1
content_length: 262
connection: keep-alive
cache_control: no-cache, no-store, max-age=0, must-revalidate
pragma: no-cache
expires: 0
strict_transport_security: max-age=31536000; includeSubDomains
x_frame_options: ALLOWALL
x_xss_protection: 1; mode=block
x_content_type_options: nosniff
referrer_policy: no-referrer
content_security_policy: default-src https:; connect-src https: wss:; font-src https: data:; frame-src https:; frame-ancestors https: http:; img-src https: data:; media-src https:; object-src https:; script-src 'unsafe-inline' 'unsafe-eval' https:; style-src 'unsafe-inline' https:;
feature_policy: payment 'none';
```

As per Open ID Spec, it can expect 400 or 401 response as per this extract from https://datatracker.ietf.org/doc/html/rfc6749#section-5.2
```
         invalid_client
               Client authentication failed (e.g., unknown client, no
               client authentication included, or unsupported
               authentication method).  The authorization server MAY
               return an HTTP 401 (Unauthorized) status code to indicate
               which HTTP authentication schemes are supported.  If the
               client attempted to authenticate via the "Authorization"
               request header field, the authorization server MUST
               respond with an HTTP 401 (Unauthorized) status code and
               include the "WWW-Authenticate" response header field
               matching the authentication scheme used by the client.
```

**Your environment**
* Edition of inferno (Community or Program): Program
* Version of inferno: 1.6.1
* Which browser and version(s) is the bug present on?: Chrome

**Steps to reproduce**
Tell us how to reproduce this issue.

Include:

* The selected test suite - https://inferno.healthit.gov/inferno/g1wjUovyon2/test_sets/test_procedure/#MultiPatientAuthorizationandAPI/test_procedure_MultiPatientAuthorizationandAPI_BulkDataAuthorizationSequence,test_procedure_MultiPatientAuthorizationandAPI_BulkDataGroupExportSequence,test_procedure_MultiPatientAuthorizationandAPI_BulkDataGroupExportValidationSequence
* The URL of the FHIR server being tested - https://fhir-qa-1.test.yourcareuniverse.net/tenant/7db7dfb5-2f73-4b96-bcf1-131410518850
* The Test ID, if it's a problem with a specific test - BDA-07 under Multiple Patient API
* Any required configuration options (client ID, client secret, etc.) - OAuth Client has been preconfigured
* Anything else needed to make the issue occur.

**Expected behavior**
Inferno should pass the test as both 400 and 401 are valid response codes as per specs listed above.

**Actual behavior**
Inferno fails the tests as shown here.
![image](https://user-images.githubusercontent.com/2487527/129791218-badde18e-adc0-4f8b-8eaf-e3551463f40d.png)

![image](https://user-images.githubusercontent.com/2487527/129791249-80c816ae-6b32-46fa-ace2-53581d2532e1.png)

Thank you for looking into it.

Regards,
Sagar Shah

**Comment by sagarshah1983** [comment]
Can we know how soon may we expect a fix on this? and which version of inferno will address this fix?

**Comment by sagarshah1983** [comment]
@yunwwang Is this resolved? I did not see any fix for this. Could you please clarify, which version would have this fix?

**Comment by yunwwang** [comment]
@sagarshah1983 sorry I put a wrong issue number in the PR. Now the PR is linked to this Issue. The label "resolved" mean that all changes are merged to main branch and waiting to be released. If you are using source code directly, you can pull the main branch and start testing with the change. If you are using Docker image or inferno.healthit.gov, you need to wait for the next release. We are planning to release next week or after.

**Comment by sagarshah1983** [comment]
Thanks for confirmation @yunwwang !

**Comment by sagarshah1983** [comment]
@yunwwang Has this been released yet? I don't see 1.7.0 up yet. Or Am I missing something here?

**Comment by yunwwang** [comment]
I am running final testing today.
